src/main/kotlin/bakalarka/elearningplatform/service/ChapterService.kt

```diff
@@ -0,0 +1,72 @@
+package bakalarka.elearningplatform.service
+
+import bakalarka.elearningplatform.db.CourseChapterRepository
+import bakalarka.elearningplatform.db.CourseRepository
+import bakalarka.elearningplatform.model.Course
+import bakalarka.elearningplatform.model.CourseChapter
+import bakalarka.elearningplatform.request.AddCourseChapterRequest
+import bakalarka.elearningplatform.security.Management
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+
+@Service
+class ChapterService(
+    var courseChapterRepository: CourseChapterRepository,
+    var courseRepository: CourseRepository,
+    var userService: UserService,
+    var management: Management
+) {
+
+    fun addChapter(request: AddCourseChapterRequest, courseId: Long): CourseChapter {
+        val (chapterTitle, description, content, position) = request
+        val course = courseRepository.findById(courseId)
+        val chapter = courseChapterRepository.save(
+            CourseChapter(
+                chapterTitle = chapterTitle,
+                description = description,
+                content = content,
+                position = position,
+                course = courseRepository.findByIdOrNull(courseId)
```

Comment:
Ahaaa, uz asi chapem, preco mas `course` nullable. Hm. Ked ale spravis to s tou exception co pisem vyssie, tak sem vies rovno poslat `course` premennu z vyssie a nemusi to byt nullable, ci? Kazdopadne - tu by sa mala pouzit premenna z hora a nie z nova volat `courseRepository` - naco?